### PR TITLE
feat: added cart functionalities

### PR DESCRIPTION
**Pull request by rahulbakshi4**
Page added :
- Cart Page

Functionalities Added:
- User can add to cart 
- User can delete item from cart 
- Cart Summary with cart quantity and cart value

### CODE REVIEW

**Review by shrey27 — APPROVED** [review]
You cart handling code looks good
Suggestions
- add some kind of alert/message on log in and sign up page
- post sign up user should redirect to homepage or last visited page, instead of login page
- Give guest credentials on both these pages


PR can be merged.

**Comment by rahulbakshi4** [comment]
> You cart handling code looks good
> Suggestions
> - add some kind of alert/message on log in and sign up page
> - post sign up user should redirect to homepage or last visited page, instead of login page
> - Give guest credentials on both these pages
> 
> 
> PR can be merged.

I'll be working on alerts and loaders for the whole project soon and add guest  credentials in next refractor.

File: src/components/Cards/Card.js [hunk 1]
```diff
@@ -5,32 +5,42 @@ import { useState, useEffect } from "react"
...
+    const [inCart, setInCart] = useState(false)
     const product = {
         _id, title, price, imageURL, rating
     }
     useEffect(() => {
         if (wishlist.wishlistItems) {
             wishlist.wishlistItems.find((item) => item._id === product._id) &&
-                setProdState({ ...prodState, inWishlist: true })
+                setInWishlist(true)
+            console.log(inWishlist)
```
Comment (ramalingaa) [inline src/components/Cards/Card.js:27]: Remove console here

File: src/index.js [hunk 2]
```diff
@@ -6,18 +6,21 @@ import { makeServer } from "./server";
 import { ProductListProvider } from "./context/product-context";
 import { AuthProvider } from './context/auth-context'
 import { WishListProvider } from "./context/wishlist-context";
+import { CartProvider } from "./context/cart-context";
```
Comment (ramalingaa) [inline src/index.js:9]: Try using index files for importing components and context

**Review by ramalingaa — APPROVED** [review]
You can reduce the number of context by using useReducer and context together.

File: src/index.js (hunk 2 above)
Comment (rahulbakshi4) [reply src/index.js:9]: Sure will do that

**Pull request merged**